aten/src/ATen/cpu/vec/vec256/vec256_int.h

```diff
@@ -731,10 +739,10 @@ class Vectorized<int8_t> : public Vectorizedi {
     return invert(_mm256_cmpgt_epi8(values, other.values));
   }
   Vectorized<int8_t> operator>(const Vectorized<int8_t>& other) const {
-    return _mm256_cmpgt_epi8(values, other.values);
+    return other < *this;
   }
   Vectorized<int8_t> operator>=(const Vectorized<int8_t>& other) const {
-    return invert(_mm256_cmpgt_epi8(other.values, values));
+    return other <= *this;
   }
```

Comment:
These can go to the base class. Even more, all this plumbing that's based on mathematical equalities (same for  `ge` and all these) should probably just be in `Vectorized`.

Reply:
So, would following do?

<details>
<summary>
Moving more functionality into `Vectorized8`
</summary>

```
template <typename T>
class Vectorized8 : public Vectorizedi {

// skipped...

public:

// skipped...

  virtual ~Vectorized8() {}

// skipped...

  virtual Vectorized<T> equal(const Vectorized8<T>& other) const = 0;
  virtual Vectorized<T> not_equal(const Vectorized8<T>& other) const = 0;
  virtual Vectorized<T> less(const Vectorized8<T>& other) const = 0;
  virtual Vectorized<T> less_or_equal(const Vectorized8<T>& other) const = 0;

  Vectorized<T> operator==(const Vectorized8<T>& other) const {
    return equal(other);
  }
  Vectorized<T> operator!=(const Vectorized8<T>& other) const {
    return not_equal(other);
  }
  Vectorized<T> operator<(const Vectorized8<T>& other) const {
    return less(other);
  }
  Vectorized<T> operator<=(const Vectorized8<T>& other) const {
    return less_or_equal(other);
  }
  Vectorized<T> operator>(const Vectorized8<T>& other) const {
    return other < *this;
  }
  Vectorized<T> operator>=(const Vectorized8<T>& other) const {
    return other <= *this;
  }

  Vectorized<T> eq(const Vectorized8<T>& other) const {
    return (*this == other) & Vectorized<T>(1);
  }
  Vectorized<T> ne(const Vectorized8<T>& other) const {
    return (*this != other) & Vectorized<T>(1);
  }
  Vectorized<T> gt(const Vectorized8<T>& other) const {
    return (*this > other) & Vectorized<T>(1);
  }
  Vectorized<T> ge(const Vectorized8<T>& other) const {
    return (*this >= other) & Vectorized<T>(1);
  }
  Vectorized<T> lt(const Vectorized8<T>& other) const {
    return (*this < other) & Vectorized<T>(1);
  }
  Vectorized<T> le(const Vectorized8<T>& other) const {
    return (*this <= other) & Vectorized<T>(1);
  }
};

template<>
class Vectorized<int8_t>: public Vectorized8<int8_t> {
public:
  using Vectorized8::Vectorized8;

  Vectorized<int8_t> neg() const;

  Vectorized<int8_t> abs() const {
    return _mm256_abs_epi8(values);
  }

  Vectorized<int8_t> equal(const Vectorized8<int8_t>& other) const override {
    return _mm256_cmpeq_epi8(values, __m256i(other));
  }
  Vectorized<int8_t> not_equal(const Vectorized8<int8_t>& other) const override {
    return invert(_mm256_cmpeq_epi8(values, __m256i(other)));
  }
  Vectorized<int8_t> less(const Vectorized8<int8_t>& other) const override {
    return _mm256_cmpgt_epi8(__m256i(other), values);
  }
  Vectorized<int8_t> less_or_equal(const Vectorized8<int8_t>& other) const override {
    return invert(_mm256_cmpgt_epi8(values, __m256i(other)));
  }
};

template<>
class Vectorized<uint8_t>: public Vectorized8<uint8_t> {
public:
  using Vectorized8::Vectorized8;

  Vectorized<uint8_t> neg() const;

  Vectorized<uint8_t> abs() const {
    return *this;
  }

  Vectorized<uint8_t> equal(const Vectorized8<uint8_t>& other) const override {
    return _mm256_cmpeq_epi8(values, __m256i(other));
  }
  Vectorized<uint8_t> not_equal(const Vectorized8<uint8_t>& other) const override {
    return invert(_mm256_cmpeq_epi8(values, __m256i(other)));
  }
  Vectorized<uint8_t> less(const Vectorized8<uint8_t>& other) const override {
    __m256i max = _mm256_max_epu8(values, __m256i(other));
    return invert(_mm256_cmpeq_epi8(max, values));
  }
  Vectorized<uint8_t> less_or_equal(const Vectorized8<uint8_t>& other) const override {
    __m256i max = _mm256_max_epu8(values, __m256i(other));
    return _mm256_cmpeq_epi8(max, __m256i(other));
  }
};
```
</details>

Reply:
Ah right, to do this properly we would need fully fledged CRTP, so perhaps it's fine to leave it as is for now.

Reply:
Indeed CRTP would be the logical progression...  So, shall we start the merge with the latest commit posted here?

Reply:
let's wait see if @peterbell10 has anything to add, but otherwise yes, this should be ready to go.